### PR TITLE
Migrate crypt mount to crypttab[SinglePass]

### PR DESCRIPTION
**Pull request by vimish**

### CODE REVIEW

File: VMEncryption/main/CryptMountConfigUtil.py [hunk 1]
```diff
@@ -610,3 +621,40 @@ def restore_mount_info(self, mount_point_or_mapper_name):
...
+
+        for crypt_item in crypt_items:
+            self.logger.log("Migrating crypt item: {0}".format(crypt_item))
+            if crypt_item.mount_point == "/" or CommonVariables.osmapper_name == crypt_item.mapper_name:
+                self.logger.log("Skipping OS disk")
+                continue
+
+            if crypt_item.mapper_name and crypt_item.mount_point and crypt_item.mount_point != "None":
+                self.logger.log(msg="Adding entry for {0} drive in fstab with mount point {1}".format(crypt_item.mapper_name, crypt_item.mount_point), level=CommonVariables.InfoLevel)
+                self.append_mount_info_data_disk(os.path.join(CommonVariables.dev_mapper_root, crypt_item.mapper_name), crypt_item.mount_point)
```
Comment (vermashi) [inline VMEncryption/main/CryptMountConfigUtil.py]: Is it feasible to quickly check here that the crypt_item.mapper_name does indeed refer to a mountable file system?

Comment (vimish) [reply VMEncryption/main/CryptMountConfigUtil.py]: @vermashi Added logic to check if mapper path exists before adding them to crypttab/fstab

Comment (vermashi) [reply VMEncryption/main/CryptMountConfigUtil.py]: I think the actual ask here was to check if the mapper device has a mountable file system.
So in addition to checking that device_path exists, add a way to test that it has a file system:
you can do one of the following:
 (1) Try to mount the device and check if it got mounted
 (2) Use disk_util.get_device_items on the device path and see if its file systems are one of the known supported file systems (xfs, ext2,3,4)

**Comment by vimish** [comment]
@vermashi Addressed review comments

File: VMEncryption/main/CryptMountConfigUtil.py [hunk 2]
```diff
@@ -610,3 +617,50 @@ def restore_mount_info(self, mount_point_or_mapper_name):
...
+            self.logger.log("Migrating crypt item: {0}".format(crypt_item))
+            if crypt_item.mount_point == "/" or CommonVariables.osmapper_name == crypt_item.mapper_name:
+                self.logger.log("Skipping OS disk")
+                continue
+
+            if crypt_item.mapper_name and crypt_item.mount_point and crypt_item.mount_point != "None":
+                self.logger.log(msg="Checking if device for mapper name: {0} has valid filesystem".format(crypt_item.mapper_name), level=CommonVariables.InfoLevel)
+                try:
+                    fstype = self.disk_util.get_device_items_property(crypt_item.mapper_name, 'FSTYPE')
+                    if fstype not in ['xfs', 'ext2', 'ext3', 'ext4']:
```
Comment (vermashi) [inline VMEncryption/main/CryptMountConfigUtil.py]: Use the list already available in CommonVariables.format_supported_file_systems

File: VMEncryption/main/CryptMountConfigUtil.py (hunk 1 above)
Comment (vimish) [reply VMEncryption/main/CryptMountConfigUtil.py]: Done

File: VMEncryption/main/CryptMountConfigUtil.py (hunk 2 above)
Comment (vimish) [reply VMEncryption/main/CryptMountConfigUtil.py]: Done.

**Comment by vermashi** [comment]
LGTM, merge when ready

**Pull request merged**